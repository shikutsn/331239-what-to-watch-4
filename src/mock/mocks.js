const appMocks = {
  promoMovie: {
    title: `The Grand Budapest Hotel`,
    genre: `Drama`,
    releaseYear: 2014,
  },
  films: [`Revenant`, `No Country for Old Men`, `Snatch`, `Bohemian Rhapsody`, `Pulp Fiction`, `Shutter Island`],
};

const testMocks = {
  promoMovie: {
    title: `No Country for Old Men`,
    genre: `Noir`,
    releaseYear: 2009,
  },
  films: [`War of the Worlds`, `Midnight Special`, `Seven Years in Tibet`, `Macbeth`, `Aviator`, `Moonrise Kingdom`],
};


export {appMocks, testMocks};
