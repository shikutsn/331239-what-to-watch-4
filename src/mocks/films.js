export default [
  {
    title: `No Country for Old Men`,
    posterSmall: `no-country-for-old-men.jpg`
  }, {
    title: `Bohemian Rhapsody`,
    posterSmall: `bohemian-rhapsody.jpg`
  }, {
    title: `Revenant`,
    posterSmall: `revenant.jpg`
  }, {
    title: `Snatch`,
    posterSmall: `snatch.jpg`
  }, {
    title: `Moonrise Kingdom`,
    posterSmall: `moonrise-kingdom.jpg`
  }, {
    title: `War of the Worlds`,
    posterSmall: `war-of-the-worlds.jpg`
  }, {
    title: `Seven Years in Tibet`,
    posterSmall: `seven-years-in-tibet.jpg`
  }, {
    title: `Johnny English`,
    posterSmall: `johnny-english.jpg`
  }
];
