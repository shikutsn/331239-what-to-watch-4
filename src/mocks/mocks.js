const appMocks = {
  promoMovie: {
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    title: `The Grand Budapest Hotel`,
    genre: `Drama`,
    releaseYear: 2014,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterSmall: `pulp-fiction.jpg`,
    rating: {
      value: 8.9,
      votesCount: 240,
    },
    description: [
      `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
      `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    ],
    director: `Wes Andreson`,
    starring: [
      `Bill Murray`,
      `Edward Norton`,
      `Jude Law`,
      `Willem Dafoe`,
    ],
  }
};

const testMocks = {
  promoMovie: {
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    title: `No Country for Old Men`,
    genre: `Noir`,
    releaseYear: 2009,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterSmall: `no-country-for-old-men.jpg`,
    rating: {
      value: 8.9,
      votesCount: 240,
    },
    description: [
      `Paragraph #1 of description for No Country for Old Men`,
      `Paragraph #2 of description for No Country for Old Men`,
    ],
    director: `director of No Country for Old Men`,
    starring: [
      `Actor #1, No Country for Old Men`,
      `Actor #2, No Country for Old Men`,
      `Actor #3, No Country for Old Men`,
      `Actor #4, No Country for Old Men`,
    ],
  },
  movies: [
    {
      background: `img/bg-the-grand-budapest-hotel.jpg`,
      title: `Pulp Fiction`,
      genre: `genre of Pulp Fiction`,
      releaseYear: 2014,
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      posterSmall: `pulp-fiction.jpg`,
      rating: {
        value: 8.9,
        votesCount: 240,
      },
      description: [
        `Paragraph #1 of description for Pulp Fiction`,
        `Paragraph #2 of description for Pulp Fiction`,
      ],
      director: `director of Pulp Fiction`,
      starring: [
        `Actor #1, Pulp Fiction`,
        `Actor #2, Pulp Fiction`,
        `Actor #3, Pulp Fiction`,
        `Actor #4, Pulp Fiction`,
      ],
    },
    {
      background: `img/bg-the-grand-budapest-hotel.jpg`,
      title: `Shutter Island`,
      genre: `genre of Shutter Island`,
      releaseYear: 2014,
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      posterSmall: `shutter-island.jpg`,
      rating: {
        value: 8.9,
        votesCount: 240,
      },
      description: [
        `Paragraph #1 of description for Shutter Island`,
        `Paragraph #2 of description for Shutter Island`,
      ],
      director: `director of Shutter Island`,
      starring: [
        `Actor #1, Shutter Island`,
        `Actor #2, Shutter Island`,
        `Actor #3, Shutter Island`,
        `Actor #4, Shutter Island`,
      ],
    },
    {
      background: `img/bg-the-grand-budapest-hotel.jpg`,
      title: `Revenant`,
      genre: `genre of Revenant`,
      releaseYear: 2014,
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      posterSmall: `revenant.jpg`,
      rating: {
        value: 8.9,
        votesCount: 240,
      },
      description: [
        `Paragraph #1 of description for Revenant`,
        `Paragraph #2 of description for Revenant`,
      ],
      director: `director of Revenant`,
      starring: [
        `Actor #1, Revenant`,
        `Actor #2, Revenant`,
        `Actor #3, Revenant`,
        `Actor #4, Revenant`,
      ],
    },
    {
      background: `img/bg-the-grand-budapest-hotel.jpg`,
      title: `Dardjeeling Limited`,
      genre: `genre of Dardjeeling Limited`,
      releaseYear: 2014,
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      posterSmall: `dardjeeling-limited.jpg`,
      rating: {
        value: 8.9,
        votesCount: 240,
      },
      description: [
        `Paragraph #1 of description for Dardjeeling Limited`,
        `Paragraph #2 of description for Dardjeeling Limited`,
      ],
      director: `director of Dardjeeling Limited`,
      starring: [
        `Actor #1, Dardjeeling Limited`,
        `Actor #2, Dardjeeling Limited`,
        `Actor #3, Dardjeeling Limited`,
        `Actor #4, Dardjeeling Limited`,
      ],
    },
    {
      background: `img/bg-the-grand-budapest-hotel.jpg`,
      title: `Moonrise Kingdom`,
      genre: `genre of Moonrise Kingdom`,
      releaseYear: 2014,
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      posterSmall: `moonrise-kingdom.jpg`,
      rating: {
        value: 8.9,
        votesCount: 240,
      },
      description: [
        `Paragraph #1 of description for Moonrise Kingdom`,
        `Paragraph #2 of description for Moonrise Kingdom`,
      ],
      director: `director of Moonrise Kingdom`,
      starring: [
        `Actor #1, Moonrise Kingdom`,
        `Actor #2, Moonrise Kingdom`,
        `Actor #3, Moonrise Kingdom`,
        `Actor #4, Moonrise Kingdom`,
      ],
    },
    {
      background: `img/bg-the-grand-budapest-hotel.jpg`,
      title: `War of the Worlds`,
      genre: `genre of War of the Worlds`,
      releaseYear: 2014,
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      posterSmall: `war-of-the-worlds.jpg`,
      rating: {
        value: 8.9,
        votesCount: 240,
      },
      description: [
        `Paragraph #1 of description for War of the Worlds`,
        `Paragraph #2 of description for War of the Worlds`,
      ],
      director: `director of War of the Worlds`,
      starring: [
        `Actor #1, War of the Worlds`,
        `Actor #2, War of the Worlds`,
        `Actor #3, War of the Worlds`,
        `Actor #4, War of the Worlds`,
      ],
    },
    {
      background: `img/bg-the-grand-budapest-hotel.jpg`,
      title: `Seven Years in Tibet`,
      genre: `genre of Seven Years in Tibet`,
      releaseYear: 2014,
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      posterSmall: `seven-years-in-tibet.jpg`,
      rating: {
        value: 8.9,
        votesCount: 240,
      },
      description: [
        `Paragraph #1 of description for Seven Years in Tibet`,
        `Paragraph #2 of description for Seven Years in Tibet`,
      ],
      director: `director of Seven Years in Tibet`,
      starring: [
        `Actor #1, Seven Years in Tibet`,
        `Actor #2, Seven Years in Tibet`,
        `Actor #3, Seven Years in Tibet`,
        `Actor #4, Seven Years in Tibet`,
      ],
    },
    {
      background: `img/bg-the-grand-budapest-hotel.jpg`,
      title: `Johnny English`,
      genre: `genre of Johnny English`,
      releaseYear: 2014,
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      posterSmall: `johnny-english.jpg`,
      rating: {
        value: 8.9,
        votesCount: 240,
      },
      description: [
        `Paragraph #1 of description for Johnny English`,
        `Paragraph #2 of description for Johnny English`,
      ],
      director: `director of Johnny English`,
      starring: [
        `Actor #1, Johnny English`,
        `Actor #2, Johnny English`,
        `Actor #3, Johnny English`,
        `Actor #4, Johnny English`,
      ],
    },
  ]};

export {appMocks, testMocks};
