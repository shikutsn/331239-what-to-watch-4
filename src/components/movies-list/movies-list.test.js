import React from "react";
import renderer from "react-test-renderer";
import MoviesList from "../movies-list/movies-list.jsx";
import {testMocks} from "../../mocks/mocks.js";

it(`Checks if movies list is rendered correctly`, () => {
  const tree = renderer
    .create(<MoviesList
      movies = {testMocks.films}
      onMovieTitleClick = {() => {}} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
