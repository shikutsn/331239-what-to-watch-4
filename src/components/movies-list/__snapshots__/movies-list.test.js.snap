// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks if movies list is rendered correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Pulp Fiction"
        height="175"
        src="img/pulp-fiction.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Pulp Fiction
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Shutter Island"
        height="175"
        src="img/shutter-island.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Shutter Island
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Revenant"
        height="175"
        src="img/revenant.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Revenant
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Midnight Special"
        height="175"
        src="img/midnight-special.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Midnight Special
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Dardjeeling Limited"
        height="175"
        src="img/dardjeeling-limited.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Dardjeeling Limited
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Orlando"
        height="175"
        src="img/orlando.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Orlando
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Mindhunter"
        height="175"
        src="img/mindhunter.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Mindhunter
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="What We Do in the Shadows"
        height="175"
        src="img/what-we-do-in-the-shadows.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        What We Do in the Shadows
      </a>
    </h3>
  </article>
</div>
`;
